openwisp_radius/api/views.py

```diff
@@ -377,17 +373,53 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
     def validate_membership(self, user):
-        if (self.organization.pk,) not in user.organizations_pk:
+        if not (user.is_superuser or (self.organization.pk,) in user.organizations_pk):
             message = _('User "{}" is not member ' 'of "{}"').format(
                 user.username, self.organization.slug
             )
             logger.warning(message)
             raise serializers.ValidationError({'non_field_errors': [message]})
 
 
+class IsAllowedRadiusBatch(BasePermission):
+    def has_permission(self, request, view):
+        radbatch_model_name = swapper.get_model_name('openwisp_radius', 'RadiusBatch')
+        app_label, _ = swapper.split(radbatch_model_name)
+        return bool(
+            request.user
+            and (request.user.is_superuser or request.user.is_staff)
+            and request.user.has_perm(f'{app_label}.view_radiusbatch')
+        )
+
+
+class DownloadRadiusBatchPdfView(DispatchOrgMixin, APIView):
+    authentication_classes = (UserTokenAuthentication, SessionAuthentication)
+    permission_classes = (IsAllowedRadiusBatch,)
+
+    def get(self, request, *args, **kwargs):
+        self.validate_membership(request.user)
+        radbatch = RadiusBatch.objects.get(pk=kwargs['radbatch'])
```

Comment:
so what happens if this line raises RadiusBatch.DoesNotExist? It doesn't seem this is catched.
Please write a test for this case as mentioned in one of the previous comments.